config/locales/en.yml

```
@@ -1438,6 +1438,11 @@ en:
       pack_by_customer: Pack By Customer
       pack_by_supplier: Pack By Supplier
       pack_by_product: Pack By Product
+      show:
+        report_taking_longer: >
+          This report is taking longer to process.
```

This seems to imply that it's still running, and leaves you unsure if you'll receive any results or not. How about:

```suggestion
          Sorry, this report took too long to process.
```

---

In fact, it is still running when background reports are enabled. I haven't implemented the timeout yet. But the user also doesn't get a result yet. So it's confusing. And without background reports, it was cancelled. I'll take your suggestion but it will be changed in my next pull request which will allow to download the report.